Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/TemporaryCacheService.java

```diff
@@ -136,7 +136,7 @@ public Storage storeProcessLog(final Integer folderId, final String logName, XLo
 		if (shouldCache(log)) {
 		    // Store corresponding object into cache
 		    cacheRepo.put(name, log);
-		    cacheRepo.put(logNameId + APMLOG_CACHE_KEY_SUFFIX, apmLogService.findAPMLogForXLog(log));
+		    cacheRepo.put(name + APMLOG_CACHE_KEY_SUFFIX, apmLogService.findAPMLogForXLog(log));
```

Comment:
To be safe, the code must handle the possible null returning result from apmLogService.findAPMLogForXLog(log) which may be caused by empty logs after validation or runtime error. Otherwise, null is stored in cache leading to possible errors in later stage of cache reading.

Reply:
It would help consistent and manageable code to have a common method called getCacheKey() to create cache key from folder id (maybe), log name and current date. This method is to be called by the three methods storeProcessLog, getProcessLog and deleteProcessLog rather than duplicate code in those methods. It would help if any changes to do the cache key in the future as well.